plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/definitionsection/StatechartDefinitionSection.java

```diff
@@ -174,15 +181,25 @@ protected Composite createDefinitionSectionComponents() {
 		return labelComposite;
 	}
 
+	protected void initSynchronizer(List<Diagram> diagramContainerHierachy) {
+		synchronizer = new DefinitionSectionSynchronizer();
+		for (Diagram diagram : diagramContainerHierachy) {
+			diagram.getElement().eAdapters().add(synchronizer);
```

Comment:
the statechart is the same for multiple subdiagrams. Only register the adapter once!! on the statechart